connect-widget/app/src/shared/api.ts

```diff
@@ -32,12 +33,13 @@ export function getApiToken(searchParams: URLSearchParams): string {
   return apiToken;
 }
 
-export function handleToken(token: string): void {
+export async function handleToken(token: string): Promise<void> {
   if (isDemoToken(token)) {
     // tell the user the widget is in demo mode && disable connect buttons
     throw new DemoTokenError();
   }
-  if (!isTokenValid()) {
+  await isTokenValid();
+  if (!(await isTokenValid())) {
```

Comment:
assuming this is a typo - regardless, this would be better for better debugging support:

```
const tokenValid = await isTokenValid();
if (!tokenValid) {
```

Reply:
Fixed